openmetrics/src/main/java/io/airlift/openmetrics/MetricsResource.java

```diff
@@ -319,7 +333,17 @@ private Stream<Metric> getManagedMetricsStream()
         Map<String, ManagedClass> managedClasses = this.mbeanExporter.getManagedClasses();
 
         return managedClasses.keySet().stream()
-                .map(objectName -> getMetricsRecursively(objectName, managedClasses.get(objectName)))
+                .map(name -> {
+                    ObjectName objectName;
+                    try {
+                        objectName = new ObjectName(name);
```

Comment:
This won't work if you use a custom ObjectNameGenerator, the ManagedClasses in jmx utils do not have object names that are guaranteed to match the jmx object name naming pattern, so this will fail for anything that implements it's own ObjectNameGenerator.

Reply:
I only do this because `mbeanExporter.getManagedClasses()` returns a map indexed by a string representation of the object name. Do you think we could change it to use objects?

Reply:
We want to also support non-jmx classes in the managedclasses from jmxutils in the future.